sdk/python/feast/registry.py

```diff
@@ -436,12 +448,24 @@ def update_registry_proto(
 
 
 class LocalRegistryStore(RegistryStore):
-    def __init__(self, repo_path: Path, registry_path_string: str):
+    def __init__(
+        self, repo_path: Path, registry_path_string: str, publish_json: bool,
+    ):
         registry_path = Path(registry_path_string)
-        if registry_path.is_absolute():
-            self._filepath = registry_path
+
+        if not registry_path.is_absolute():
+            registry_path = repo_path.joinpath(registry_path)
+
+        if registry_path_string[-1] == "/":
```

Comment:
Since file nor dir hasn't been created yet, there's no way to check if the `registry_path_string` passed in is intended to be dir or file. Here I infer that if it ends in `/` then it should be dir.

Reply:
What if we looked at the extension? we can assume if its `blahblah.something` then its a file, and everything else is a folder, like `blahblah` and `blahblah/` would be folders.

Reply:
I had thought about that too - the tradeoff is that if their folder names have a single dot (not sure if this is a large enough edge case), then there would be no way for us to infer their intention. In addition, do you know if all filepaths/filenames require an extension? Happy to make this change if you've thought about these things too!

Reply:
Filenames don't require an extension, but our default registry path is `data/registry.db`, so the majority of users will have an extension. I think it's the least bad approach tbh. I don't think there are folks creating folders with an extension. @tsotnet @achals any thoughts?

Reply:
I agree with @woop  - in the case that the registry file/dir has to be created, the presence or absence of an extension should be good enough for us to determine whether this should be a file or a dir.

Reply:
Sounds good, I'm cool with that.